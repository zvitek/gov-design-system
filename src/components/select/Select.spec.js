import { shallowMount } from '@vue/test-utils'
import GovSelect from '@components/select/Select'

let wrapper

describe('GovSelect', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovSelect)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovSelect')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        const wrapper = shallowMount(GovSelect, {
            propsData: {
                customUniqueId: 'select_abc'
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('has a select element', () => {
        expect(wrapper.contains('select')).toBeTruthy()
    })

    it('emit input event with value when value change', () => {
        wrapper.setProps({ value: true })
        expect(wrapper.vm.computedValue).toBeTruthy()
        wrapper.vm.computedValue = false
        const valueEmitted = wrapper.emitted()['input'][0]
        expect(valueEmitted).toContainEqual(false)
    })
})
