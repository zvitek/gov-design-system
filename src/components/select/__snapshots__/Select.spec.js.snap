// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovSelect render correctly 1`] = `
<div class="control"><span class="select is-empty"><select><!----> </select></span>
    <!---->
</div>
`;
